authentication fix

bitnik opened the pull request:
This PR fixes the check if authenticated user has already a running server (''). And also updates required JupyterHub version.

choldgraf commented:
Thanks for the fix!

Is this something testable? Was it a bug or is this just a more explicit way of testing for a running server?

bitnik commented:
this is better way to inform users if they already have a running server and they try to launch a new binder (this is for the case if named servers are not enabled and each user can only launch one binder at a time). otherwise they get a "Failed to launch image ..." message.

Actually I already implemented it in this way, but after [this review](https://github.com/jupyterhub/binderhub/pull/666/files/a6235c0ec44fa7ab6ae70d84b8c9430890e19bb4#diff-28deb80e5e4fbd96b8b1bdcd3ce1cc3eR148) (the comment is wrong, it actually checks if server name is empty), I changed it wrongly [in this commit](https://github.com/jupyterhub/binderhub/pull/666/commits/3d228a8e5aa52130920f8e66b6c2fb8024b0ba2d#diff-28deb80e5e4fbd96b8b1bdcd3ce1cc3eR148).
